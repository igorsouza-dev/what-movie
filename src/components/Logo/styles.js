import styled from 'styled-components';

export const Title = styled.h1`
  font-family: 'Oleo Script', cursive;
  font-size: ${props => props.size}px;
  color: ${props => props.color};
  width: 100%;
  text-align: center;
`;
