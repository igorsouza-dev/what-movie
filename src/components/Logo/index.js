import React from 'react';
import PropTypes from 'prop-types';
import { colors } from 'styles/variables';
import { Title } from './styles';

function Logo({ size, color }) {
  return (
    <Title size={size} color={color} data-testid="logo">
      What Movie?
    </Title>
  );
}

Logo.propTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
};
Logo.defaultProps = {
  size: 80,
  color: colors.logo,
};

export default Logo;
