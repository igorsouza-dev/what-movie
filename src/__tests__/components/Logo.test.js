import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Logo from 'components/Logo';

describe('<Logo />', () => {
  afterEach(cleanup);
  it('should render', () => {
    const { container, getByTestId } = render(<Logo />);

    const linkElement = getByTestId(/logo/i);
    expect(linkElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render correclty', () => {
    const { container, getByText } = render(<Logo />);

    const linkElement = getByText(/What Movie?/i);
    expect(linkElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
