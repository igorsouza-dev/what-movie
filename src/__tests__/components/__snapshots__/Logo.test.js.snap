// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render 1`] = `
<h1
  class="sc-bdVaJa lhpWri"
  color="#d5dcf9"
  data-testid="logo"
  size="80"
>
  What Movie?
</h1>
`;

exports[`<Logo /> should render correclty 1`] = `
<h1
  class="sc-bdVaJa lhpWri"
  color="#d5dcf9"
  data-testid="logo"
  size="80"
>
  What Movie?
</h1>
`;
